template/registry/datetime.go

```diff
@@ -11,129 +11,185 @@ import (
 // https://github.com/unicode-org/message-format-wg/blob/20a61b4af534acb7ecb68a3812ca0143b34dfc76/spec/registry.xml#L13
 
 // datetimeFunc is the implementation of the datetime function. Locale-sensitive date and time formatting.
-var datetimeRegistryFunc = &Func{
-	Name:           "datetime",
-	Func:           datetimeFunc,
-	MatchSignature: nil, // Not allowed to use in matching context
-	FormatSignature: &Signature{
-		IsInputRequired: true,
-		ValidateInput: func(a any) error {
-			if _, ok := a.(time.Time); !ok {
-				return fmt.Errorf("unsupported type: %T", a)
+var datetimeRegistryFunc = F{
+	Format: datetimeFunc,
+}
+
+type datetimeOptions struct {
+	TimeZone               *time.Location
+	DateStyle              string
+	TimeStyle              string
+	Calendar               string
+	NumberingSystem        string
+	HourCycle              string
+	DayPeriod              string
+	Weekday                string
+	Era                    string
+	Year                   string
+	Month                  string
+	Day                    string
+	Hour                   string
+	Minute                 string
+	Second                 string
+	TimeZoneName           string
+	FractionalSecondDigits int
+}
+
+func parseDatetimeInput(input any) (time.Time, error) {
+	if input == nil {
+		return time.Time{}, errors.New("input is required, got nil")
+	}
+
+	v, ok := input.(time.Time)
+	if !ok {
+		return time.Time{}, fmt.Errorf("unsupported type: %T", input)
+	}
+
+	return v, nil
+}
+
+func parseDatetimeOptions(options Opts) (*datetimeOptions, error) {
+	var (
+		opts datetimeOptions
+		err  error
+	)
+
+	// The predefined date formatting style to use.
+	dateStyles := []string{"full", "long", "medium", "short"}
+	if opts.DateStyle, err = options.GetString("dateStyle", "", dateStyles); err != nil {
```

Comment:
No sensible fallback values?

Reply:
https://github.com/unicode-org/message-format-wg/blob/main/spec/registry.xml#L31 no default value defined in MF2 WG.